pkg/plugins/external/helpers.go

```diff
@@ -102,8 +105,52 @@ func makePluginRequest(req external.PluginRequest, path string) (*external.Plugi
 	return &res, nil
 }
 
+// getUniverseMap is a helper function that is used to read the current directory to build
+// the universe map.
+// It will return a map[string]string where the keys are relative paths to files in the directory
+// and values are the contents, or an error if an issue occured while reading one of the files.
+func getUniverseMap(fs machinery.Filesystem) (map[string]string, error) {
```

Comment:
Question: Why have the path for the files is helpful? Should we not pass here the PROJECT config instead?

Reply:
@camilamacedo86 - the way I see this is that when passing the path of files, we preserve the universe's file system structure, this way, the external plugin will have access to all scaffolded files the same way it would if we were to scan the root directory of the project from the external plugin.
on other hand, if there were multiple scaffolded files in the universe with the same name but in different directories, using the path of the files as keys will prevent overriding the file content in the universe map since they will have different keys.

Example: 
plugin chain = plugin-1/v1 -> plugin-2/v1
plugin-1 scaffolds `dir-1/doc.go` and `dir-2/doc.go`
by using the file path as map key, we make sure that both entries for `dir-1/doc.go` and `dir-2/doc.go` will be available in the universe map, because the keys are unique, as opposed to if we use something like file name as the key, we will end up overriding one of the entries.
so in `plugin-2/v1`, we can for example access and make changes to any of the scaffolds the same way we would access them from file system: i.e. `pr.Universe["dir-1/doc.go"]`

Reply:
Thank you for the explanation.